server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -680,6 +674,7 @@ public void insert(InsertPlan insertPlan) throws QueryProcessException {
       }
       insertPlan.setDataTypes(dataTypes);
       storageEngine.insert(insertPlan);
+      insertPlan.updateMNodeLastValues(node);
```

Comment:
It is not suitable to put the updateMNodeLastValues method in the InsertPlan.
You can update the last value in the before loop, there you already got the measurementNode.

Reply:
The update function is moved into MNode structure.
I added a loop to update the last value after storageEngine.insert() instead, since it is possible that insert() could fail and throw exceptions. In such case the last value cache should not be updated.